regolith/helpers/u_finishprumhelper.py

```diff
@@ -70,16 +70,27 @@ def db_updater(self):
             print("Please rerun the helper specifying the complete ID.")
             return
         found_projectum.update({'status':'finished'})
-        if rc.pub_sum:
-            found_projectum['public_summary'] = rc.pub_sum
-        elif found_projectum.get('public_summary') is not None:
+
+        if found_projectum['professional_summary']:
```

Comment:
I think the logic is wrong here.  It may overwrite pub_sum if prof_sum and rc.pub_sum.

THis is a good chance for you to handle your test driven development.

1.  make a function and call the function here.  Call it `sync_db_cli` maybe
2.  write the docstring that explains what logic the function should have.  This results in a docstrng but also focusses your  mind on the logic you want.
3. write a test_sync_db_cli function.  You can put it in this same file for now, but later it will to to somewhere else in the test suite
4. write a bunch of inputs and asserts that capture the logic
5. make sure you think about some edge cases (things of the wrong type passed in, etc)
5. run the tests and make sure they all fail (they will because initially the function just has `pass` in it.
6. write some logic and run the tests and keep doing it until they all pass.  You can write logic so they pass one at a time, or if you have an idea for a more global solution, have them all pass at once.

1. 

make a function that we will test.  Call the function here

1.

Reply:
Umm, I think the logic should work according to our yesterday discussion. I handled all the four conditions in my codes. If both prof_sum and rc.pub_sum, it will just raise error and ask user to type again.
Yup sure, I will make a test function.